### DISCUSSION TITLE
Documentation de "une possibilité"

### DISCUSSION THREAD

**Discussion by laem** [publicodes/publicodes#429] [Ideas]
Je vois que ce mécanisme qui n'en n'est pas un est marqué comme "va évoluer bientôt" dans betagouv/publicodes#27 
Mais en attendant, j'ai l'impression qu'il est nulle part dans la doc, or c'est une brique assez importante du langage. 

Notamment le fait de devoir spécifier une valeur par défaut de ce style me semble vraiment pas évident.

```yaml
par défaut: "'tulipe'"
```
Je pense qu'il faut l'ajouter, qu'en pensez-vous ?

**Comment by laem** [comment]
https://github.com/betagouv/ecolab-data/commit/ae177efe99b54fd740c7d8cff0c9b9ad50fdccea
Pas très beau 😅 en tout cas ça liste notre utilisation chez ecolab de "une possibilité".

**Comment by johangirod** [comment]
Effectivement... Mais depuis que `par défaut` est évalué et peut référencer une autre règle il fallait différencier le cas de la chaîne de caractère.

Pour la doc, j'avoue être un peu embêté, je ne sais pas trop quoi dire. La logique d'évaluation du mécanisme est pour l'instant vide : https://github.com/betagouv/mon-entreprise/blob/master/publicodes/source/mecanisms/one-possibility.ts

Toute la logique derrière est dans `RuleInput.ts`, d'où le fait que la doc publicodes core reste vide pour l'instant. Lorsque une possibilité codera pour un type Enum, on aura plus de chose à dire. Ça devrait venir bientôt.

**Comment by laem** [comment]
OK je vois le souci. Mais j'ai l'impression qu'on peut quand même utiliser ce mécanisme dans publicodes sans génération de simulateur.

Par exemple, pour [ce petit bout de code](https://bit.ly/3adaYxV).

Là, dans le studio, on n'en fera rien parce qu'on ne peut pas spécifier à la fois la formule "une possibilité" (qui en l'absence de la génération du formulaire ne sert qu'à typer les valeurs possibles de "motorisation") et une valeur de simulation, par exemple "hybride". 

Mais dans l'utilisation en vrai de publicodes, même si c'est une coquille vide, le mécanisme permet quand même de générer une documentation du typage de "motorisation", même si effectivement pour le calcul, la seule chose exécutable c'est le `variations` qui au passage sert aussi de typage, mais potentiellement partiel.

**Comment by johangirod** [comment]
Le mécanisme a été supprimé de publicodes. Il reviendra lorsque nous travaillerons sur un type option.

**Comment by laem** [comment]
@johangirod marrant, je viens juste de tomber ici via une recherche et plusieurs sauts. 

Je dois justement intégrer une question à choix multiples dans ma nouvelle app, et je n'ai rien trouvé dans la doc. 

Ça vaut le coup je pense de documenter le sujet dans https://publi.codes/docs/guides/créer-formulaire. 

Je vais itérer de mon côté d'abord.

**Comment by laem** [comment]
Bon, je confirme que coder le système de "une / plusieurs possibiltés", qu'on appelle mosaïque chez NGC @Clemog @EmileRolley , c'est pas trivial. 

Au moins aussi compliqué que de faire le lot input text, nombre et binaires.

Car ça revient à faire des groupes de questions. 

D'où l'intérêt d'avoir un template :) J'ai toujours pas de code à ouvrir pour l'instant, mais ça viendra. Faut juste que je stoppe à temps avant que mon app devienne trop spécialisée.

**Comment by Clemog** [comment]
> Bon, je confirme que coder le système de "une / plusieurs possibiltés", qu'on appelle mosaïque chez NGC @Clemog @EmileRolley , c'est pas trivial.

On a une carte pour justement proposé d'ajouter dans la doc de Publicodes notre manière de gérer les mosaïques, ça reste spécifique mais ça pourrait être repris !
